crates/figma_import/src/toolkit_style.rs

```diff
@@ -179,13 +180,59 @@ pub struct ProgressMarkerMeterData {
     pub end_y: f32,
 }
 
+// Schema for progress vector data that we read from Figma plugin data
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct ProgressVectorMeterDataSchema {
+    pub enabled: bool,
+    pub discrete: bool,
+    pub discrete_value: f32,
+    pub paths: Vec<figma_schema::Path>,
+}
+
+// Schema for progress vector data that we write to serialized data
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct ProgressVectorMeterData {
+    pub enabled: bool,
+    pub discrete: bool,
+    pub discrete_value: f32,
+    pub paths: Vec<vector_schema::Path>,
+}
+
+// Schema for dials & gauges Figma plugin data
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub enum MeterDataSchema {
+    ArcData(ArcMeterData),
+    RotationData(RotationMeterData),
+    ProgressBarData(ProgressBarMeterData),
+    ProgressMarkerData(ProgressMarkerMeterData),
+    ProgressVectorData(ProgressVectorMeterDataSchema),
+}
+
+// Schema for dials & gauges data that we write to serialized data
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub enum MeterData {
     ArcData(ArcMeterData),
     RotationData(RotationMeterData),
     ProgressBarData(ProgressBarMeterData),
     ProgressMarkerData(ProgressMarkerMeterData),
+    ProgressVectorData(ProgressVectorMeterData),
+}
+impl From<MeterDataSchema> for MeterData {
```

Comment:
Fwiw now that I think about it, MeterData probably belongs in the node definition (because it's functional) rather than the style definition (which is more presentation stuff), but it's OK to leave it like this for now.

Reply:
I am planning to refactor this a bit during the `dc_bundle` and protobuf migration.